src/main/java/uk/antiperson/stackmob/config/MainConfig.java

```diff
@@ -1,59 +1,372 @@
+    private final Set<EntityType> default_drops_type_blacklist = new ObjectOpenHashSet<>();
+
+
+    private boolean default_experience_enabled;
+    private final Map<EntityType, Boolean> experience_enabled = new Object2ObjectOpenHashMap<>();
+    private double default_experience_multiplier_min;
+    private final Map<EntityType, Double> experience_multiplier_min = new Object2ObjectOpenHashMap<>();
+    private double default_experience_multiplier_max;
+    private final Map<EntityType, Double> experience_multiplier_max = new Object2ObjectOpenHashMap<>();
+    private final Set<EntityType> default_experience_type_blacklist = new ObjectOpenHashSet<>();
+
+
+    private boolean default_player_stats;
+    private final Map<EntityType, Boolean> player_stats = new Object2ObjectOpenHashMap<>();
+
+
+    private boolean default_wait_to_stack_enabled;
+    private final Map<EntityType, Boolean> wait_to_stack_enabled = new Object2ObjectOpenHashMap<>();
+    private int default_wait_to_stack_wait_time;
+    private final Map<EntityType, Integer> wait_to_stack_wait_time = new Object2ObjectOpenHashMap<>();
+    private final Set<CreatureSpawnEvent.SpawnReason> default_wait_to_stack_reasons_whitelist = new ObjectOpenHashSet<>();
+    private final Map<EntityType, Set<CreatureSpawnEvent.SpawnReason>> wait_to_stack_reasons_whitelist = new Object2ObjectOpenHashMap<>();
+    private final Set<EntityType> default_wait_to_stack_types_whitelist = new ObjectOpenHashSet<>();
+
+
+    private boolean default_disable_targeting_enabled;
+    private final Map<EntityType, Boolean> disable_targeting_enabled = new Object2ObjectOpenHashMap<>();
+    private final Set<EntityType> default_disable_targeting_type_blacklist = new ObjectOpenHashSet<>();
+    private final Set<CreatureSpawnEvent.SpawnReason> default_disable_targeting_reason_blacklist = new ObjectOpenHashSet<>();
+    private final Map<EntityType, Set<CreatureSpawnEvent.SpawnReason>> disable_targeting_reason_blacklist = new Object2ObjectOpenHashMap<>();
+
+
+    private final Set<EntityType> default_types_blacklist = new ObjectOpenHashSet<>();
+    private final Set<CreatureSpawnEvent.SpawnReason> default_reason_blacklist = new ObjectOpenHashSet<>();
+    private final Map<EntityType, Set<CreatureSpawnEvent.SpawnReason>> reason_blacklist = new Object2ObjectOpenHashMap<>();
+    private final Set<World> default_worlds_blacklist = new ObjectOpenHashSet<>();
+    private final Map<EntityType, Set<World>> worlds_blacklist = new Object2ObjectOpenHashMap<>();
+
+
+    private final Map<String, Boolean> default_events_remove_stack_data = new Object2ObjectOpenHashMap<>();
+    private final Map<EntityType, Map<String, Boolean>> events_remove_stack_data = new Object2ObjectOpenHashMap<>();
+    private final Map<String, ListenerMode> default_events_mode = new Object2ObjectOpenHashMap<>();
+    private final Map<EntityType, Map<String, ListenerMode>> events_mode = new Object2ObjectOpenHashMap<>();
+
```

Comment:
Irritating point of my changes, you're right, but I swear config checks are heavy and this is the better solution I personnaly found to support custom settings and general options with good performances :/
We could put these changes aside and we'll see later for a new config system...

Reply:
I could also suggest doing config additions as you want (like in the current system), and let me work on the optimized version later, because I designed this and I perfectly know how it should work :)